src/devices/Qmc5883l/Qmc5883l.cs

```diff
@@ -0,0 +1,150 @@
+
+        /// <summary>
+        /// Typical Data Output Rate (Hz).
+        /// </summary>
+        /// <value>10Hz by default.</value>
+        public OutputRate OutputRate { get; set; } = OutputRate.Rate10Hz;
+
+        /// <summary>
+        /// Bandwidth of an internal digital filter.
+        /// </summary>
+        /// <value>256 by default.</value>
+        public Oversampling Oversampling { get; set; } = Oversampling.Rate256;
+
+        /// <summary>
+        /// The field range goes hand in hand with the sensitivity of the magnetic sensor.
+        /// </summary>
+        /// <value>8 Gauss by default.</value>
+        public FieldRange FieldRange { get; set; } = FieldRange.Gauss8;
+
+        /// <summary>
+        /// Sensors operation mode.
+        /// </summary>
+        /// <value>In Standby mode by default. Make sure to change it to Continuous before trying to get any data!</value>
+        public Mode DeviceMode { get; set; } = Mode.Standby;
+
+        /// <summary>
+        /// QMC5883L Direction Vector
+        /// </summary>
+        [Telemetry]
+        public Vector3 GetDirection() => _isDeviceInitialized ? ReadDirectionVector() : throw new IOException("Sensor has not yet been initialized. Please call SetMode() to initialize it.");
+
+        /// <summary>
+        /// QMC5883L Heading (DEG)
+        /// </summary>
+        /// <returns>Heading(Angle)</returns>
+        public Angle GetHeading() => VectorExtension.GetHeading(GetDirection());
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Qmc5883l"/> class.
+        /// </summary>
+        public Qmc5883l(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+        }
+
+        /// <summary>
+        /// Sets the sensors mode.
+        /// </summary>
+        public void SetMode()
+        {
+            Span<byte> interruptCommand = stackalloc byte[]
+            {
+                (byte)Registers.QMC_CONFIG_REG_2_ADDR,
+                (byte)RollPointer
+            };
+            Span<byte> resetCommand = stackalloc byte[]
+            {
+                (byte)Registers.QMC_RESET_REG_ADDR,
+                0x01
+            };
+            byte config = (byte)((byte)DeviceMode | (byte)OutputRate | (byte)FieldRange | (byte)Oversampling);
+            Span<byte> setMainRegisteryCommand = stackalloc byte[]
+            {
+                (byte)Registers.QMC_CONFIG_REG_1_ADDR,
+                config
+            };
+            _i2cDevice.Write(interruptCommand);
+            _i2cDevice.Write(resetCommand);
+            _i2cDevice.Write(setMainRegisteryCommand);
+            _isDeviceInitialized = true;
+        }
+
+        /// <summary>
+        /// Reads status register to determine if data is avaliable.
+        /// </summary>
+        /// <returns>State of the register as a boolean</returns>
+        public bool IsReady()
```

Comment:
You should rename to `IsDataReady`. My first thought was that this was about the device being ready for usage